leases/schema/mutations.py

```diff
@@ -189,6 +189,71 @@ def mutate_and_get_payload(cls, root, info, **input):
         return DeleteBerthLeaseMutation()
 
 
+class SwitchBerthMutation(graphene.ClientIDMutation):
+    class Input:
+        old_lease_id = graphene.ID(required=True)
+        new_berth_id = graphene.ID(required=True)
+        switch_date = graphene.Date(
+            description="The date which will mark the end of the old lease and the start of the new lease."
+            "If none is provided, it will default to the day when the mutation is called"
+        )
+
+    old_berth_lease = graphene.Field(BerthLeaseNode)
+    new_berth_lease = graphene.Field(BerthLeaseNode)
+
+    @classmethod
+    @add_permission_required(BerthLease)
+    @change_permission_required(BerthLease)
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, old_lease_id, new_berth_id, **input):
+        switch_date = input.get("switch_date") or calculate_berth_lease_start_date()
```

Comment:
Should add some sanity checking for switch_date? I guess it should be possible to have switch_date in the past, in case admins just forgot to record a switch in the db(?). My guess is switch_date should be max 6 months in the past - maybe verify with Joonas?

Reply:
Let's verify.

Reply:
Verified that 6 months is a good threshold.